### PR TITLE
Adds annotations for generated daemonset pods

### PR DESCRIPTION
**Pull request by meik99**
# Description

This change adds a `Annotation` field to the `HostInjectSpec`.
It allows specifying annotations to be added to the pods of oneagent daemonset.

## How can this be tested?

* Deploy a dynakube with cloudnative, classic, or host monitoring enabled and the annotations field set
* When deployed, the pods must have the default apparmor annotation that is always applied and additionally the custom annotation

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/kubeobjects/labels.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (labels *AppLabels) BuildMatchLabels() map[string]string {
 	}
 }
 
-func MergeLabels(labels ...map[string]string) map[string]string {
+func MergeMap(labels ...map[string]string) map[string]string {
```
Comment (0sewa0) [inline src/kubeobjects/labels.go:113]:
```suggestion
func MergeMap(maps ...map[string]string) map[string]string {
```

Comment (0sewa0) [reply src/kubeobjects/labels.go:113]: (it could also use generics 😛 )

Comment (meik99) [reply src/kubeobjects/labels.go]: done

File: src/kubeobjects/labels.go [hunk 2]
```diff
@@ -110,9 +110,9 @@ func (labels *AppLabels) BuildMatchLabels() map[string]string {
 	}
 }
 
-func MergeLabels(labels ...map[string]string) map[string]string {
+func MergeMap(maps ...map[string]string) map[string]string {
```
Comment (0sewa0) [inline src/kubeobjects/labels.go:113]: there is a `map.go` file, this function would fit it perfectly 🥲

Comment (meik99) [reply src/kubeobjects/labels.go:113]: :smiling_face_with_tear:

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**